File: ruby/lib/warehouse_actions/replenish_stock_command.rb
require_relative '../products/catalogue'
require_relative '../user_interface/user_input'
require_relative '../warehouse/warehouse'

module WarehouseActions

	class ReplenishStockCommand

		def initialize(warehouse, catalogue)
			@warehouse = warehouse
			@catalogue = catalogue
		end

		def run(cmd)
			if (cmd.args.length != 5)
				System.err.println("ERROR: Usage: r aisle loc sku num")
				return
			end
			if (@catalogue.lookup(cmd.args(3)) == null)
				throw new UnknownProductException(cmd.args(3))
			end
			sku = [cmd.args(1), cmd.args(2), cmd.args(3)]
			begin
				int numItems = Integer.parseInt(cmd.args(4))
				@warehouse.replenish(sku, numItems)
			rescue (NumberFormatException e)
				System.err.println("ERROR: Usage: r sku numitems")
			end
		end

	end
end

File: python3/warehouse_actions/replenish_stock_command.py
import sys
from . unknown_product_exception import UnknownProductException

class ReplenishStockCommand:

    def __init__(self, warehouse, catalogue):
        self.warehouse = warehouse
        self.catalogue = catalogue

    def run(self, cmd):
        if len(cmd.args) != 5:
            print("ERROR: Usage: r aisle loc sku num", file=sys.stderr)
            return
        if not self.catalogue.lookup(cmd.arg(3)): raise UnknownProductException(cmd.arg(3))
        sku = [cmd.arg(1), cmd.arg(2), cmd.arg(3)]
        numItems = int(cmd.arg(4))
        self.warehouse.replenish(sku, numItems)

